sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosClientTelemetryConfig.java

```diff
@@ -48,8 +57,19 @@ public final class CosmosClientTelemetryConfig {
     private String clientCorrelationId = null;
     private EnumSet<TagName> metricTagNamesOverride = null;
     private boolean isClientMetricsEnabled = false;
+    private final HashSet<CosmosDiagnosticsHandler> customDiagnosticHandlers;
+    private final CopyOnWriteArrayList<CosmosDiagnosticsHandler> diagnosticHandlers;
```

Comment:
why do we need CopyOnWriteArrayList? This will have too much perf impact if it falls anywhere under the main code path.

Reply:
This needs to be threadsafe - also iterating over it - and changes to diagnostics handlers are extremely rare - like only when you add at runtime a new handler - and CopyOnWriteArrayList is providing significantly better perf for reads / iterations compared to ConcurrentLinkedQueue/Dequeue as long as writes are happening rarely.